test/unit/test_id_storage.c

```diff
@@ -0,0 +1,187 @@
+	bool r;
+	ut32 id0, id1, id2;
+	RIDStorage *ids = r_id_storage_new (1, 15);
+	r = r_id_storage_add (ids, "aaa", &id0);
+	mu_assert ("id0", r);
+	mu_assert ("id0", id0 == 1);
+	r = r_id_storage_add (ids, "bbb", &id1);
+	mu_assert ("id1", r);
+	mu_assert ("id1", id1 == 2);
+	r = r_id_storage_add (ids, "ccc", &id2);
+	mu_assert ("id2", r);
+	mu_assert ("id2", id2 == 3);
+	ut32 id = id1;
+	r_id_storage_get_prev (ids, &id);
+	r_id_storage_get_next (ids, &id);
+	r_id_storage_free (ids);
+	
+	mu_assert_eq (id, id1, "r_id_storage_{next/prev} reversal 0");
+	mu_end;
+}
+
+static bool test_r_id_storage_prev_next_eq_1(void) {
+	// test if next reverts prev (modulo wrap 1)
+	bool r;
+	ut32 id, id0, id1, id2;
+	RIDStorage *ids = r_id_storage_new (0, 15);
+	r_id_storage_add (ids, "bbb", &id0);
+	r_id_storage_add (ids, "aaa", &id1);
+	r_id_storage_add (ids, "ccc", &id2);
+	id = id0;
+	r = r_id_storage_get_prev (ids, &id);
+	mu_assert ("get_prev(0) doesnt exist", !r);
+	ut32 n = id;
+	id = id1;
+	r = r_id_storage_get_prev (ids, &id);
+	mu_assert ("get_prev(1) must exist", r);
+	r = r_id_storage_get_next (ids, &id);
+	mu_assert ("get_prev(1) must exist", r);
+	r_id_storage_free (ids);
+	
+	mu_assert_eq (id, id1, "r_id_storage_{next/prev} reversal 1");
+	mu_end;
+}
+
+static bool test_r_id_storage_prev_next_min(void) {
+	// test if next reverts prev (modulo wrap 1)
+	bool r;
+	ut32 id, id0, id1, id2;
+	RIDStorage *ids = r_id_storage_new (3, 15);
+	r_id_storage_add (ids, "bbb", &id0);
+	r_id_storage_add (ids, "aaa", &id1);
+	r_id_storage_add (ids, "ccc", &id2);
+	id = id0;
+	r = r_id_storage_get_prev (ids, &id);
+	mu_assert ("get_prev(0) doesnt exist", !r);
+	ut32 n = id;
+	id = id1;
+	r = r_id_storage_get_prev (ids, &id);
+	mu_assert ("get_prev(1) must exist", r);
+	r = r_id_storage_get_next (ids, &id);
+	mu_assert ("get_prev(1) must exist", r);
+	r_id_storage_free (ids);
+	mu_assert_eq (id, id1, "r_id_storage_{next/prev} reversal 1");
+	mu_end;
+}
+
+static bool test_r_id_storage_empty(void) {
+	// test if next reverts prev (modulo wrap 2)
+	ut32 id, _id;
+	RIDStorage *ids = r_id_storage_new (0, 15);
+	bool r = r_id_storage_get_prev (ids, &_id);
+	mu_assert ("get prev from none", !r);
+	r = r_id_storage_get_next (ids, &_id);
+	mu_assert ("get next from none", !r);
+	r = r_id_storage_get_highest (ids, &_id);
+	mu_assert ("get high from none", !r);
+	r = r_id_storage_get_lowest (ids, &_id);
+	mu_assert ("get low from none", !r);
+	r_id_storage_free (ids);
+	
+	mu_end;
+}
+
+static bool test_r_id_storage_prev_next_eq_2(void) {
+	// test if next reverts prev (modulo wrap 2)
+	ut32 id, _id;
+	RIDStorage *ids = r_id_storage_new (0, 15);
+	r_id_storage_add (ids, "aaa", &_id);
+	r_id_storage_add (ids, "ccc", &_id);
+	r_id_storage_add (ids, "bbb", &id);
+	_id = id;
+	r_id_storage_get_prev (ids, &_id);
+	r_id_storage_get_next (ids, &_id);
+	r_id_storage_free (ids);
+	
+	mu_assert_eq (id, _id, "r_id_storage_{next/prev} reversal 2");
+	mu_end;
```

Comment:
this is again like all the other tests i've mentioned before.. we are always testing the exact same code paths, I think.